### DISCUSSION TITLE
CoApp Installation not recognized

### DISCUSSION THREAD

**Discussion by mrtootoo** [aclap-dev/video-downloadhelper#1111] [Help and Q&A] [answered]
### Your question

Installed CoApp, not recognized.  Install and diagnostic details below:
Ubuntu Mate
Release 22.04.4 LTS (Jammy Jellyfish) 64-bit
Kernel Linux 5.15.0-117-generic x86_64
MATE 1.26.0
Firefox 128.0.3 (64-bit)

```
~$ curl -sSLf https://github.com/aclap-dev/vdhcoapp/releases/latest/download/install.sh | bash
Downloading: https://github.com/aclap-dev/vdhcoapp/releases/latest/download/vdhcoapp-linux-x86_64.tar.bz2
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 52.8M  100 52.8M    0     0  20.6M      0  0:00:02  0:00:02 --:--:-- 27.8M
Extracting tarball…
/tmp/vdhcoapp-ppt6Pa: bzip2 compressed data, block size = 900k
Registering CoApp
Installing…
Flatpak is installed. Making the coapp available from browser sandboxes:
Linked coapp within org.mozilla.firefox.
Linked coapp within com.brave.Browser.
Linked coapp within com.google.Chrome.
Linked coapp within com.google.ChromeDev.
Linked coapp within org.chromium.Chromium.
Linked coapp within com.github.Eloston.UngoogledChromium.
Linked coapp within com.microsoft.Edge.
Linked coapp within com.microsoft.EdgeDev.
Writing /home/mate/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json
/home/mate/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json : Cannot write manifest file: EACCES: permission denied, open '/home/mate/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json'
CoApp successfuly installed under '~/.local/share/vdhcoapp'.
To uninstall, run '~/.local/share/vdhcoapp/vdhcoapp uninstall' and remove '~/.local/share/vdhcoapp'.
Re-run that script to update the coapp.

$ /opt/vdhcoapp/vdhcoapp --info
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/opt/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/mate",
  "converterBinary": "/opt/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360"
}


### Addon & CoApp version and details

version: 9.1.0.42
target: mozilla
channel: stable
lang: en-US
coapp: {"found":false,"error":"No such native application net.downloadhelper.coapp"}
license: {"unneeded":true}
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:128.0) Gecko/20100101 Firefox/128.0
{
    "download_directory": "/home/mate/Downloads",
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 5
}
```

**Comment by Mattmjs** [comment]
Did you refer to the Wiki before posting here:  https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-not-recognized#ubuntu-and-flatpak--snap-based-systems

**Reply by mrtootoo** [reply]
I did.  I ran the commands and posted above (and below).  There seems to be an error:  "Cannot write manifest file: EACCES: permission denied"  Am I missing something?

$ ~/.local/share/vdhcoapp/vdhcoapp install
Installing…
Flatpak is installed. Making the coapp available from browser sandboxes:
Linked coapp within org.mozilla.firefox.
Linked coapp within com.brave.Browser.
Linked coapp within com.google.Chrome.
Linked coapp within com.google.ChromeDev.
Linked coapp within org.chromium.Chromium.
Linked coapp within com.github.Eloston.UngoogledChromium.
Linked coapp within com.microsoft.Edge.
Linked coapp within com.microsoft.EdgeDev.
Writing /home/mate/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json
/home/mate/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json : Cannot write manifest file: EACCES: permission denied, open '/home/mate/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json'
$ /opt/vdhcoapp/vdhcoapp install
Installing…
Flatpak is installed. Making the coapp available from browser sandboxes:
Linked coapp within org.mozilla.firefox.
Linked coapp within com.brave.Browser.
Linked coapp within com.google.Chrome.
Linked coapp within com.google.ChromeDev.
Linked coapp within org.chromium.Chromium.
Linked coapp within com.github.Eloston.UngoogledChromium.
Linked coapp within com.microsoft.Edge.
Linked coapp within com.microsoft.EdgeDev.
Writing /home/mate/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json
/home/mate/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json : Cannot write manifest file: EACCES: permission denied, open '/home/mate/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json'

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I also don't know diddly squat about Linux.  But if I saw error messages like that on Windows, I'd be scrambling to figure out the ownership status of the containing directory in question.  I'd be looking at the permissions & the other weird things Windows does with security.  If Linux is anything at all like Windows in this area, there's a bit of a fancy dance you have to perform to make things writeable.

Based on stuff I've read in this forum over the years, here & on the old forum on Google Groups, I've gathered that there is a difference between system level & user level in Linux.  I suppose the analogous thing in Windows is administrator vs ordinary user.  Is that a hint for you?  My system is captive.  I'm the only user & I'm the administrator.  I do everything at the administrator level & my administrator user owns every directory on every partition.  Windows 11 has this concept that a computer can be used by multiple users & areas of the disk space can be segregated so each user can see only its own data.  I've reduced this world-view to a single user.  Does any of this babbling give you any ideas?

**Reply by mrtootoo** [reply]
Thanks for the reply.  I understand what you are saying.  What confuses me is that the installation instructions are very clear "(not as root! Don't use sudo)".  My understanding is that root or sudo is needed to elevate permissions.  So until I get more feedback, I am resistant to install as root.

**Comment by paulrouget** [comment]
Do **NOT** use root.

Type these commands please:

```
id
```

then:

```
ls -ld /home/mate/.mozilla
ls -ld /home/mate/.mozilla/native-messaging-hosts
ls -ld /home/mate/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json
```

And tell me what you get.

**Reply by mrtootoo** [reply]
It seems "net.downloadhelper.coapp.json" is missing.  The only file in "/native-messaging-hosts/" is "com.add0n.node.json"

$ sudo id
uid=0(root) gid=0(root) groups=0(root)

$ sudo ls -ld /home/mate/.mozilla
drwx------ 6 mate mate 4096 Jun  6  2020 /home/mate/.mozilla

$ sudo ls -ld /home/mate/.mozilla/native-messaging-hosts
drwxr-xr-x 2 root root 4096 Jun  6  2020 /home/mate/.mozilla/native-messaging-hosts

$ sudo ls -ld /home/mate/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json
ls: cannot access '/home/mate/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json': No such file or directory

**Comment by paulrouget** [comment, marked as answer]
So sorry, I initially wrote "do use root", and I mean "do NOT use root" …

Anyway, the issue is that your directory has the wrong rights. To fix it, do this:

```
sudo chown -R mate.mate /home/mate/.mozilla/native-messaging-hosts
```

**Comment by mrtootoo** [comment]
Thank you for your patience and assiatnce

**Comment by GrampaWildWilly** [comment]
So it seems I was kind of right.  Only problem is that I couldn't give the advice in Linux-speak, like Paul did.  I speak Windows & could have given the advice in Windows-speak, which doesn't help you.

Just to improve my minimal understanding, I see a string that looks like this:

drwxr-xr-x

I gather that indicates that the specified directory (the d) has Read/Write/eXecute permissions for  . . . who?  Why are there 3 sets of permissions?  I can see that 2 of the sets omit Write permission, which fits perfectly with the error message you were getting.  But one of the sets does specify Write permission.  Why wasn't that good enough?
